server/routes/getValidatorNamesAndAddresses.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable guard-for-in */
+import { Request, Response, NextFunction } from 'express';
+import { Op } from 'sequelize';
+
+const getValidatorNamesAndAddresses = async (models, req: Request, res: Response, next: NextFunction) => {
+
+    try {
+        //fetch address of validators
+        let validators = await models.Validators.findAll({
+            attributes: ['stash'],
+        });
+        let validatorAddresses = validators.map(obj => obj.stash);
+
+        //fetch address of validators
+        let profile = await models.OffchainProfile.findAll({
+            attributes: ['data'],
+            include: [{
+                model: models.Address,
+                attributes: ['address'],
+                where: {
+                    address: {
+                        [Op.in]: validatorAddresses,
+                    }
+                },
+                required: true,
+            }]
+        });
+
+        const profileData = profile?.map(element => {
+            element.data = JSON.parse(element.data);
+            let setProfile = { name: element.data.name, address: element.Address?.address };
+            return setProfile;
+        });
+        //set names of corresponding addresses in validators
```

Comment:
Please remove this logic, we should not be updating any data in a GET request. This is an anti-pattern.